### PR TITLE
tkt-71587: Fix erroneous template messages on creation failure

### LINKED ISSUE #809: Can't re-use jail names when creating with a template
Make sure to follow and check these boxes before submitting an issue! Thank you.

- [x] Supply `iocage --version`

# iocage --version
Version	1.0 RELEASE 2018/12

- [x] Supply the commands used, along with any steps to recreate it.
- [x] Provide the output from the command you issued.
- [x] Supply what you expected the result or output to be
- [x] Checked that the problem has not already been fixed on `master` if using
 a stable release.

# iocage create -t base -n foo
foo successfully created!

# iocage destroy foo

This will destroy jail foo

Are you sure? [y/N]: y
Destroying foo

# iocage create -t base -n foo
Template: foo not found!

### PR DESCRIPTION
**Pull request by skarekrow**
FreeNAS Ticket: #71587
Closes #809

### CODE REVIEW

File: iocage_cli/create.py [hunk 1]
```diff
@@ -179,20 +180,23 @@ def cli(release, template, count, props, pkglist, basejail, thickjail, empty,
                       template=template, short=short, _uuid=_uuid,
                       basejail=basejail, thickjail=thickjail, empty=empty,
                       thickconfig=thickconfig)
-    except RuntimeError as err:
-        if template and "Dataset" in str(err):
+    except (RuntimeError, ioc_exceptions.JailMissingConfiguration) as err:
+        if template and "Dataset" in str(err) or str(
+                err).startswith('Template'):
```
Comment (sonicaj) [inline iocage_cli/create.py:185]: 8 spaces

Comment (sonicaj) [reply iocage_cli/create.py:185]: Should we perhaps do a case insensitive comparison ?

**Review by sonicaj — APPROVED** [review]
LGTM, just a couple of notes whether we should look at those right now

File: iocage_cli/create.py (hunk 1 above)
Comment (skarekrow) [reply iocage_cli/create.py:185]: 8 spaces are for flake, since we're line breaking with `str(` and no, the message will be capitalized.

**Comment by skarekrow** [comment]
Unrelated test failure

**Pull request merged**